Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -388,22 +442,25 @@ internal async void ClearAllPoolsAsync()
 
         public void SetMaxPoolSize(int size)
         {
-            _maxPoolSize = size;
+            _poolConfig.MaxPoolSize = size;
+            _configOverriden = true;
         }
 
         public int GetMaxPoolSize()
         {
-            return _maxPoolSize;
+            return _poolConfig.MaxPoolSize;
         }
 
-        public void SetTimeout(long time)
+        public void SetTimeout(long seconds)
         {
-            _timeout = time;
+            var timeout = seconds < 0 ? TimeoutHelper.Infinity() : TimeSpan.FromSeconds(seconds);
+            _poolConfig.ExpirationTimeout = timeout;
+            _configOverriden = true;
```

Comment:
It never gets cleared which makes me think that keeping values altered by setters and reflecting it connection string could be potentially better. On the other hand we've decided not to have it since connections would have to be moved from one pool to another after such change.
To decide - maybe parsing connection string and using a portion of it as a pool key which will allow to keep in sync pool config values with its connection string and at the same time not to move connections between the pools when setting a tech value happens. WDYT @sfc-gh-knozderko @sfc-gh-dstempniak @sfc-gh-pbulawa 
Of course if only, let's have another PR for such approach